src/main/java/nextstep/fp/Lambda.java

```diff
@@ -27,30 +27,19 @@ public void run() {
     }
 
     public static int sumAll(List<Integer> numbers) {
-        int total = 0;
-        for (int number : numbers) {
-            total += number;
-        }
-        return total;
+        return sumAll(numbers, number -> true);
     }
 
     public static int sumAllEven(List<Integer> numbers) {
-        int total = 0;
-        for (int number : numbers) {
-            if (number % 2 == 0) {
-                total += number;
-            }
-        }
-        return total;
+        return sumAll(numbers, number -> number % 2 == 0);
     }
 
     public static int sumAllOverThree(List<Integer> numbers) {
-        int total = 0;
-        for (int number : numbers) {
-            if (number > 3) {
-                total += number;
-            }
-        }
-        return total;
+        return sumAll(numbers, number -> number > 3);
+    }
+
+    public static int sumAll(List<Integer> numbers,
+                          Conditional c) {
+        return numbers.stream().filter(number->c.test(number)).mapToInt(i -> i).sum();
     }
```

Comment:
```suggestion
    public static int sumAll(List<Integer> numbers, Conditional conditional) {
        return numbers.stream()
                .filter(conditional::test)
                .mapToInt(Integer::intValue)
                .sum();
    }
```

변수를 축약하지 않고 한줄에 점을 하나만을 찍는다면 코드의 가독성이 좋아지지 않을까요?